kolibri/core/assets/src/views/PaginatedListContainer.vue

```diff
@@ -74,65 +74,105 @@
         required: false,
         default: 30,
       },
+      totalPageNumber: {
+        type: Number,
+        required: false,
+        default: 1,
+      },
+      roleFilter: {
+        type: Object,
+        required: false,
+        default: null,
+      },
+      excludeMemberOf: {
+        type: String,
+        required: false,
+        default: '',
+      },
+      userAssignmentType: {
+        type: String,
+        required: false,
+        default: '',
+      },
     },
     data() {
       return {
         filterInput: '',
         currentPage: 1,
+        currentPageNumber: 1,
+        userList: this.items,
+        totalPageNumbers: this.totalPageNumber,
+        totalItems: this.items.length,
       };
     },
     computed: {
-      filteredItems() {
-        return filterUsersByNames(this.items, this.filterInput);
-      },
       numFilteredItems() {
-        return this.filteredItems.length;
-      },
-      totalPages() {
-        return Math.ceil(this.numFilteredItems / this.itemsPerPage);
-      },
-      startRange() {
-        return (this.currentPage - 1) * this.itemsPerPage;
+        return this.totalPageNumbers;
       },
       visibleStartRange() {
-        return Math.min(this.startRange + 1, this.numFilteredItems);
+        return this.currentPageNumber;
       },
       endRange() {
         return this.currentPage * this.itemsPerPage;
       },
       visibleEndRange() {
         return Math.min(this.endRange, this.numFilteredItems);
       },
-      visibleFilteredItems() {
-        return this.filteredItems.slice(this.startRange, this.endRange);
-      },
       previousButtonDisabled() {
-        return this.currentPage === 1 || this.numFilteredItems === 0;
+        return this.currentPageNumber === 1 || this.numFilteredItems === 0;
       },
       nextButtonDisabled() {
         return (
-          this.totalPages === 1 ||
-          this.currentPage === this.totalPages ||
+          this.totalPageNumbers === 1 ||
+          this.currentPageNumber === this.totalPageNumbers ||
           this.numFilteredItems === 0
         );
       },
     },
     watch: {
-      visibleFilteredItems: {
```

Comment:
I would prefer to keep this interface of emitting page changes and deferring data loading to the place that is using the `PaginatedListContainer`. At the moment data loading is happening in both the `kolibri/plugins/facility/assets/src/modules/classAssignMembers/handlers.js` file and in here.

The other issue is that this will now break the other uses of this component.

I would suggest adding a prop to the component called `allItems` that indicates whether all the items have been passed (as is the current usage) or whether it is a pre-paginated subset, as you are now implementing. We can't merge this in its current state as it will break functionality in multiple other parts of the app.

Reply:
Thanks, @rtibbles for your review. I will look into these issues and make the required changes.

Reply:
I think we can use prop `isPaginated` to check whether we have to run the old code or the new one. So we have to make the change in `UserPage` component only and it will have no effect on other pages.